src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,94 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMA = ",";
+    private static final String SPACE = " ";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String fileContent = readFileContent(fromFileName);
+        String[] splitText = splitTextIntoWords(fileContent);
+        int[] results = calculateResult(splitText);
+        String report = createReport(results);
+        writeContentToFile(report, toFileName);
+    }
+
+    private String readFileContent(String fromFileName) {
+        StringBuilder allText = new StringBuilder();
+        File file = new File(fromFileName);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
```

Comment:
The 'File' class is used here, but since Java 7, it's recommended to use 'Path' and 'Files' for file operations for better exception handling and interoperability.